src/services/codefixes/fixMissingTypeAnnotationOnExports.ts

```diff
@@ -0,0 +1,1192 @@
+    Diagnostics.Variable_must_have_an_explicit_type_annotation_with_isolatedDeclarations.code,
+    Diagnostics.Parameter_must_have_an_explicit_type_annotation_with_isolatedDeclarations.code,
+    Diagnostics.Property_must_have_an_explicit_type_annotation_with_isolatedDeclarations.code,
+    Diagnostics.Expression_type_can_t_be_inferred_with_isolatedDeclarations.code,
+    Diagnostics.Binding_elements_can_t_be_exported_directly_with_isolatedDeclarations.code,
+    Diagnostics.Computed_properties_must_be_number_or_string_literals_variables_or_dotted_expressions_with_isolatedDeclarations.code,
+    Diagnostics.Enum_member_initializers_must_be_computable_without_references_to_external_symbols_with_isolatedDeclarations.code,
+    Diagnostics.Extends_clause_can_t_contain_an_expression_with_isolatedDeclarations.code,
+    Diagnostics.Objects_that_contain_shorthand_properties_can_t_be_inferred_with_isolatedDeclarations.code,
+    Diagnostics.Objects_that_contain_spread_assignments_can_t_be_inferred_with_isolatedDeclarations.code,
+    Diagnostics.Arrays_with_spread_elements_can_t_inferred_with_isolatedDeclarations.code,
+    Diagnostics.Default_exports_can_t_be_inferred_with_isolatedDeclarations.code,
+    Diagnostics.Only_const_arrays_can_be_inferred_with_isolatedDeclarations.code,
+    Diagnostics.Assigning_properties_to_functions_without_declaring_them_is_not_supported_with_isolatedDeclarations_Add_an_explicit_declaration_for_the_properties_assigned_to_this_function.code,
+    Diagnostics.Declaration_emit_for_this_parameter_requires_implicitly_adding_undefined_to_it_s_type_This_is_not_supported_with_isolatedDeclarations.code,
+    Diagnostics.Add_satisfies_and_a_type_assertion_to_this_expression_satisfies_T_as_T_to_make_the_type_explicit.code,
+];
+
+const canHaveTypeAnnotation = new Set<SyntaxKind>([
+    SyntaxKind.GetAccessor,
+    SyntaxKind.MethodDeclaration,
+    SyntaxKind.PropertyDeclaration,
+    SyntaxKind.FunctionDeclaration,
+    SyntaxKind.FunctionExpression,
+    SyntaxKind.ArrowFunction,
+    SyntaxKind.VariableDeclaration,
+    SyntaxKind.Parameter,
+    SyntaxKind.ExportAssignment,
+    SyntaxKind.ClassDeclaration,
+    SyntaxKind.ObjectBindingPattern,
+    SyntaxKind.ArrayBindingPattern,
+]);
+
+const declarationEmitNodeBuilderFlags = NodeBuilderFlags.MultilineObjectLiterals
+    | NodeBuilderFlags.WriteClassExpressionAsTypeLiteral
+    | NodeBuilderFlags.UseTypeOfFunction
+    | NodeBuilderFlags.UseStructuralFallback
+    | NodeBuilderFlags.AllowEmptyTuple
+    | NodeBuilderFlags.GenerateNamesForShadowedTypeParams
+    | NodeBuilderFlags.NoTruncation
+    | NodeBuilderFlags.WriteComputedProps;
+
+enum TypePrintMode {
+    // Prints it's fully spelled out type
+    FULL,
+    // Prints a relative type i.e. typeof X
+    RELATIVE,
+    // Prints a widened type in case the expression is known to
+    // e.g. export const a = Math.random() ? "0" : "1"; the type will be `string` in d.ts files
+    WIDENED,

```

Comment:
Nit, but we typically just use `PascalCase` for enum members, not UPPERCASE like enums in other languages

Reply:
Oh right, done.